packages/kbn-i18n/src/core/i18n.ts

```diff
@@ -177,25 +209,25 @@ export function translate(
     defaultMessage: '',
   }
 ) {
+  const shouldUsePseudoLocale = isPseudoLocale();
+
   if (!id || !isString(id)) {
     throw new Error('[I18n] An `id` must be a non-empty string to translate a message.');
   }
 
-  const message = getMessageById(id);
+  const message = shouldUsePseudoLocale ? defaultMessage : getMessageById(id);
 
   if (!message && !defaultMessage) {
     throw new Error(`[I18n] Cannot format message: "${id}". Default message must be provided.`);
   }
 
-  if (!hasValues(values)) {
```

Comment:
__nit:__ I know I suggested that before, but I've just looked into `intl-messageformat` and I think it's too expensive to call format for all labels even if they don't contain values :/ Let's keep this `if`, but change it to `!hasValue(values) && !shouldUsePseudoLocale` with proper comment. Pseudo locale is supposed to be run in dev env only, so calling `format` there is fine, but not in prod.